assets/javascripts/kitten/components/carousel/carousel.stories.js

```diff
@@ -0,0 +1,145 @@
+import React from 'react'
+import { storiesOf } from '@storybook/react'
+import { withKnobs, text, number, object } from '@storybook/addon-knobs'
+import { CrowdfundingCard } from '../../components/cards/crowdfunding-card'
+import { Carousel } from '../../components/carousel/carousel'
+import { CONTAINER_PADDING } from '../../constants/grid-config'
+import { Marger } from '../../components/layout/marger'
+import { Container } from '../../components/grid/container'
+import { Grid, GridCol } from '../../components/grid/grid'
+
+const StoryContainer = ({ children }) => (
+  <Container>
+    <Marger top="5" bottom="5">
+      {children}
+    </Marger>
+  </Container>
+)
+
+const SmallStoryContainer = ({ children }) => (
+  <Container>
+    <Marger top="5" bottom="5">
+      <Grid>
+        <GridCol col="10" offset="1">
+          <Grid>
+            <GridCol col="8" offset="2">
+              {children}
+            </GridCol>
+          </Grid>
+        </GridCol>
+      </Grid>
+    </Marger>
+  </Container>
+)
+
+const data = [
+  {
+    title: 'Item A',
+    imageSrc: 'http://placekitten.com/500/300',
+    thumbSrc: 'http://placekitten.com/80/80',
+  },
+  {
+    title: 'Item B',
+    imageSrc: 'http://placekitten.com/501/301',
+    thumbSrc: 'http://placekitten.com/81/81',
+  },
+  {
+    title: 'Item C',
+    imageSrc: 'http://placekitten.com/502/302',
+    thumbSrc: 'http://placekitten.com/82/82',
+  },
+  {
+    title: 'Item D',
+    imageSrc: 'http://placekitten.com/503/303',
+    thumbSrc: 'http://placekitten.com/83/83',
+  },
+  {
+    title: 'Item E',
+    imageSrc: 'http://placekitten.com/504/304',
+    thumbSrc: 'http://placekitten.com/84/84',
+  },
+]
+
+const paginationPosition = {
+  default: 'bottom',
+  fromL: 'left',
+}
+
+storiesOf('Carousel/ProjectCarousel', module)
+  .addDecorator(withKnobs)
+  .add('default', () => (
+    <StoryContainer>
+      <Carousel
+        itemMinWidth={number('itemMinWidth', 280)}
+        baseItemMarginBetween={number(
+          'baseItemMarginBetween',
+          CONTAINER_PADDING,
+        )}
+        data={data}
+        paginationPosition={object('paginationPosition', paginationPosition)}
+        renderItem={({ item }) => (
```

Comment:
Je ne sais pas ce que vous en pensez mais quit à faire un *breaking change*, ça pourrait valoir le coup de rendre les `Item` comme des `props.children` et s'éviter de mettre les props `data` et `renderItem` et faire directement `<Carousel props>{data.map(data => <Card data/>)}</Carousel`
l'API en gagnerait en simplicité !

Reply:
Yes carrément. Je pense que ca peut valoir le coup d'adopter la même stratégie que les boutons en faisant un composant un nouveau carrousel. Ca évite le breaking change sur les carrousels déjà en place, ca permet de plus facilement itérer sur nos besoins et de faire évoluer l'API.